driver-core/src/main/com/mongodb/operation/OperationHelper.java

```diff
@@ -77,6 +79,30 @@ public Void apply(final T t) {
         }
     }
 
+    static void checkValidReadConcern(final Connection connection, final ReadConcern readConcern) {
+        if (!serverIsAtLeastVersionThreeDotTwo(connection.getDescription()) && !readConcern.equals(ReadConcern.DEFAULT)) {
+            throw new IllegalArgumentException(format("Unsupported ReadConcern : '%s'", readConcern.asDocument().toJson()));
+        }
+    }
+
+    static void checkValidReadConcern(final AsyncConnection connection, final ReadConcern readConcern,
+                                      final AsyncCallableWithConnection callable) {
+        Throwable throwable = null;
+        if (!serverIsAtLeastVersionThreeDotTwo(connection.getDescription()) && !readConcern.equals(ReadConcern.DEFAULT)) {
```

Comment:
Might as well use ReadConcern.isServerDefault here as well, as in the operations.